core/jobs.py

```diff
@@ -1041,16 +1290,15 @@ def _kickoff_batch_job_after_previous_one_ends(cls):
 
     @classmethod
     def on_batch_job_completion(cls):
-        """Called when a batch job completes."""
+        """Marks itself complete and prepares for next job."""
         job_status = cls._process_job_completion_and_return_status()
         if job_status == job_models.CONTINUOUS_COMPUTATION_STATUS_CODE_RUNNING:
             cls._kickoff_batch_job_after_previous_one_ends()
 
     @classmethod
     def on_batch_job_canceled(cls):
+        """Marks as idle because the job should already be stopping."""
```

Comment:
Marks the batch job as canceled and verifies that the continuous computation is now idle.